reference/camera.md

```diff
@@ -520,3 +527,64 @@ images, the `quality` parameter is ignored.
+
+{[C++](cpp-api.md#cpp_camera)}, {[Java](java-api.md#java_camera)}, {[Python](python-api.md#python_camera)}, {[Matlab](matlab-api.md#matlab_camera)}, {[ROS](ros-api.md)}
+
+```c
+#include <webots/camera.h>
+
+bool wb_camera_has_recognition(WbDeviceTag tag);
+void wb_camera_recognition_enable(WbDeviceTag tag, int sampling_period);
+void wb_camera_recognition_disable(WbDeviceTag tag);
+int wb_camera_recognition_get_sampling_period(WbDeviceTag tag);
+int wb_camera_recognition_get_number_of_objects(WbDeviceTag tag);
+const WbCameraRecognitionObject *wb_camera_recognition_get_objects(WbDeviceTag tag);
+```
+
+**Description**
+
+If a [Recognition](recognition.md) node is present in the `recognition` field, the camera can recognize objects in its image.
+
+The `wb_camera_has_recognition` function can be used to determine whether a [Recognition](recognition.md) node is present or not.
+
+The `wb_camera_recognition_enable()` function allows the user to enable the recognition, note that it is not needed to enable the camera to let recognition work.
+
+The `wb_camera_recognition_disable()` function turns off the recognition, saving computation time.
+
+The `wb_camera_recognition_get_sampling_period()` function returns the period given to the `wb_camera_recognition_enable()` function, or 0 if the recognition is disabled.
+
+The `wb_camera_recognition_get_number_of_objects` and `wb_camera_recognition_get_objects` functions allow the user to get the current number of recognized objects and the objects array.
+
+---
+
+**Camera recognition object**
+
+A camera recognition object is defined by the following structure:
+
+```c
+typedef struct {
+ int      id;
+ double   position[3];
+ double   orientation[4];
+ double   size[2];
+ int      position_on_image[2];
+ int      size_on_image[2];
+ int      number_of_colors;
+ double  *colors;
+ char    *model;
+} WbCameraRecognitionObject;
+```
+
+The `id` represents the node id corresponding to the object, it is possible to use this id directly in the [wb_supervisor_node_get_from_id](supervisor.md#wb_supervisor_node_get_from_def) supervisor function. The `position` and `orientation` are expressed relatively to the camera (the relative position is the one of the center of the object which can differ from its origin) and the units are meter and radian. The `size` represents the X and Y sizes in meters relatively to the camera (it is of course impossible to know the depth of the object). The `position_on_image` and `size_on_image` can be used to determine the bounding box of the object in the camera image, the units are pixels. The `number_of_colors` and `colors` returns respectively the number of colors of the objects and pointer to the colors array, each colors is represented by 3 doubles (R, G and B), therefore the size of the array is equal to 3 * `number_of_colors`. Finally `model` returns the `model` field of the [Solid](solid.md) node.
```

Comment:
I would add "and" before "it is possible" on the first line.

"each colors" -> "each color"